include/circt/Dialect/LLHD/IR/LLHDStructureOps.td

```diff
@@ -86,6 +86,46 @@ def FinalOp : LLHDOp<"final", [
   let assemblyFormat = "attr-dict-with-keyword $body";
 }
 
+//===----------------------------------------------------------------------===//
+// CombinationalOp
+//===----------------------------------------------------------------------===//
+
+def CombinationalOp : LLHDOp<"combinational", [
+  NoRegionArguments,
+  RecursiveMemoryEffects,
+  HasParent<"hw::HWModuleOp">,
+]> {
+  let summary = "A process that runs when any of its operand values change";
+  let description = [{
+    An `llhd.combinational` op encapsulates a region of IR that executes
+    whenever any of the values used in its body change. Control flow must
+    eventually end in an `llhd.halt` terminator. The process may have results,
+    in which case the `llhd.halt` terminators must provide a list of values to
+    yield for the process results.
```

Comment:
I think it's worth mentioning that the halt operation here does not permanently halt the process like it does in an `llhd.process` but just suspends it until the next value change is detected. It is essentially just like an `llhd.wait` in an `llhd.process` where the sensitivity list implicitly contains all values used in the region that are defined outside of it.

Reply:
Great idea. I've tweaked this to use an `llhd.yield` terminator, and reworded the description.